### PR TITLE
feat(workflow): u#1540 t#4000 create workflow endpoint

### PR DESCRIPTION
**Pull request by tdelatorre**
## Create workflow endpoint

This PR:
- Add workflows to get_project endpoint
- Updated list workflows ordered by oldest first
- Add create workflow endpoint

### How to test
- [x] Review code
- [x] Tests are green

Import postman files
- [x] Create a new workflow -> 200. Return a workflow with its default statuses
- [x] Create a new workflow with the same name -> slug has an incremental suffix
- [x] Create a new workflow with not_exist project -> 404 Not found
- [x] Create a new workflow with an user without permissions -> 403 Forbidden
- [x] Create a new workflow with empty name -> 422 Unprocessable Entity
- [x] Create a new workflow with name > 40 -> 422 Unprocessable Entity
- [x] Try to create a new workflow when you have already 8 workflows created -> 400 Bad Request
- [x] List workflows -> ordered by oldest first
- [x] Get project -> has workflows

### CODE REVIEW

File: python/apps/taiga/src/taiga/projects/projects/repositories.py [hunk 1]
```diff
@@ -166,7 +166,7 @@ def list_projects(
 def get_project(
     filters: ProjectFilters = {},
     select_related: ProjectSelectRelated = ["workspace"],
-    prefetch_related: ProjectPrefetchRelated = ["workspace"],
+    prefetch_related: ProjectPrefetchRelated = ["workspace", "workflows"],
```
[the remark this answers is not in the document]
Comment (tdelatorre) [reply python/apps/taiga/src/taiga/projects/projects/repositories.py]: Done!

File: python/apps/taiga/src/taiga/projects/projects/models.py [hunk 2]
```diff
@@ -99,6 +99,7 @@ class ProjectTemplate(models.BaseModel):
     slug = models.LowerSlugField(max_length=250, null=False, blank=True, unique=True, verbose_name="slug")
     roles = models.JSONField(null=True, blank=True, verbose_name="roles")
     workflows = models.JSONField(null=True, blank=True, verbose_name="workflows")
+    workflow_statuses = models.JSONField(null=True, blank=True, verbose_name="workflow_statuses")
```
[the remark this answers is not in the document]
Comment (tdelatorre) [reply python/apps/taiga/src/taiga/projects/projects/models.py]: Done!

File: python/apps/taiga/src/taiga/workflows/api/__init__.py [hunk 3]
```diff
@@ -22,11 +22,13 @@
     ReorderWorkflowStatusesValidator,
     UpdateWorkflowStatusValidator,
     WorkflowStatusValidator,
+    WorkflowValidator,
```
[the remark this answers is not in the document]
Comment (tdelatorre) [reply python/apps/taiga/src/taiga/workflows/api/__init__.py]: Done!

File: python/apps/taiga/src/taiga/workflows/api/validators.py [hunk 4]
```diff
@@ -15,19 +15,29 @@
...
     max_length = 30
 
 
+class WorkflowName(ConstrainedStr):
+    strip_whitespace = True
+    min_length = 1
+    max_length = 40
+
+
+class WorkflowValidator(BaseModel):
```
[the remark this answers is not in the document]
Comment (tdelatorre) [reply python/apps/taiga/src/taiga/workflows/api/validators.py]: Done!

File: python/apps/taiga/src/taiga/workflows/services/__init__.py [hunk 5]
```diff
@@ -45,6 +88,21 @@ async def list_workflows(project_id: UUID) -> list[WorkflowSerializer]:
...
+    )
+
+    return [
+        serializers_services.serialize_workflow_nested(
+            workflow=workflow,
+        )
+        for workflow in workflows
+    ]
+
+
```
[the remark this answers is not in the document]
Comment (tdelatorre) [reply python/apps/taiga/src/taiga/workflows/services/__init__.py]: Done!

File: python/apps/taiga/src/taiga/workflows/serializers/services.py [hunk 6]
```diff
@@ -23,6 +23,13 @@ def serialize_workflow(workflow: Workflow, workflow_statuses: list[WorkflowStatu
     )
 
 
+def serialize_workflow_nested(workflow: Workflow) -> WorkflowNestedSerializer:
```
Comment (bameda) [inline python/apps/taiga/src/taiga/workflows/serializers/services.py:26]: Creo que esta función ya no la usa nadie

**Review by bameda — APPROVED** [review]

File: python/apps/taiga/src/taiga/workflows/serializers/services.py (hunk 6 above)
Comment (tdelatorre) [reply python/apps/taiga/src/taiga/workflows/serializers/services.py]: Done!

**Pull request merged**